tests/test_plotting.py

```diff
@@ -0,0 +1,99 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import tempfile
+import os
+
+import lsst.utils.tests
+import lsst.utils
+
+from lsst.summit.utils.plotting import plot
+from lsst.summit.utils.utils import detectObjectsInExp
+import lsst.afw.image as afwImage
+from lsst.afw.geom import SpanSet
+from lsst.afw.detection import Footprint
+
+
+class PlottingTestCase(lsst.utils.tests.TestCase):
+    afwDataDir = lsst.utils.getPackageDir('afwdata')
+    filename = 'postISRCCD_2020021800224-EMPTY~EMPTY-det000.fits.fz'
+
+    @classmethod
+    def setUpClass(cls):
+        cls.outputDir = tempfile.mkdtemp()
+
+    def test_plot(self):
+        """Test that the the plot is made and saved
+        """
+        fullName = os.path.join(self.afwDataDir, "LATISS/postISRCCD", self.filename)
+        exp = afwImage.ExposureF(fullName)
+        centroids = [(567, 746), (576, 599), (678, 989)]
+
+        foot1 = Footprint(SpanSet.fromShape(5, offset=(690, 710)))
+        foot2 = Footprint(SpanSet.fromShape(6, offset=(159, 216)))
+        fpset = detectObjectsInExp(exp)
+
+        # Input is an exposure
+        outputFilename = os.path.join(self.outputDir, 'testPlotting_exp.jpg')
+        plot(exp,
+             centroid=centroids,
+             footprints=fpset,
+             showCompass=True,
+             addLegend=True,
+             savePlotAs=outputFilename)
+        self.assertTrue(os.path.isfile(outputFilename))
+        self.assertTrue(os.path.getsize(outputFilename) > 10000)
+
+        # Input is a numpy array
+        outputFilename = os.path.join(self.outputDir, 'testPlotting_nparr.jpg')
+        nparr = exp.image.array
+        plot(nparr,
+             footprints=foot1,
+             savePlotAs=outputFilename)
+        self.assertTrue(os.path.isfile(outputFilename))
+        self.assertTrue(os.path.getsize(outputFilename) > 10000)
+
+        # Input is an image
+        outputFilename = os.path.join(self.outputDir, 'testPlotting_image.jpg')
+        im = exp.image
+        plot(im,
+             footprints=[foot1, foot2],
+             savePlotAs=outputFilename)
+        self.assertTrue(os.path.isfile(outputFilename))
+        self.assertTrue(os.path.getsize(outputFilename) > 10000)
+
+        # Input is a masked image
+        outputFilename = os.path.join(self.outputDir, 'testPloting_mask.jpg')
+        masked = exp.maskedImage
+        plot(masked,
+             savePlotAs=outputFilename)
+        self.assertTrue(os.path.isfile(outputFilename))
+        self.assertTrue(os.path.getsize(outputFilename) > 10000)
+
```

Comment:
I think it might be worth adding a loop to check that the various stretching options work. You don't need to save and check every plot, I don't think, but something that tests all the stretch options at least would be a good addition I think.

Reply:
Maybe manipulate one of the pixels in the image to set it to nan, too, just to check that everything is nan-safe too.

Reply:
I set a small area of the image to nans and added code to test stretches.

Reply:
That's nearly it, but `None` is not the same as `NaN`. Please set them to `np.nan` so that the nan-safety can be properly checked.

Reply:
Also, actually, this doesn't check that it works with all-nan images, and I think that is something that needs to at least not crash, and at present that is indeed a crash because the colorbar can't handle the limits. Once that's fixed I think this is basically good to merge.

Reply:
How do you want to proceed here? Exit with an error that nans are in the image or give a warning that nans are in the image and they will be replaced with something that the code can handle?

Reply:
If the image is all Nan there is no point in plotting. The code will check if image is pure nan and will raise ValueError.

Reply:
Some nans in the image is perfectly acceptable, and the code should (and I think already does?) handle it fine, I'm just saying that there should be a test for it.

For all-nan images there is no point in plotting, but it happens, and this code should give an all-white image rather than raising. You could do a quick check for if the entire array is nan if you like and then just return a blank image, perhaps. At this point I'm happy for you to just file another ticket for that, assign it to me, and add a TODO in the code though.

Reply:
Part of this is already in the code (check if image is all nan). There is also a test for cases with few nans in the image. It is included in test for plotting.py. I will change the ValueError part to blank image.

Reply:
Yup, I just checked and that all seems to be working perfectly now, thanks.